pkg/scheduler/framework/listers.go

```diff
@@ -40,3 +45,75 @@ type SharedLister interface {
 	NodeInfos() NodeInfoLister
 	StorageInfos() StorageInfoLister
 }
+
+// ResourceSliceLister is used by the DRA plugin to obtain ResourceSlices.
+type ResourceSliceLister interface {
+	// List returns a list of all ResourceSlices.
+	List() ([]*resourceapi.ResourceSlice, error)
+}
+
+// DeviceClassLister is used by the DRA plugin to obtain DeviceClasses.
+type DeviceClassLister interface {
+	// List returns a list of all DeviceClasses.
+	List() ([]*resourceapi.DeviceClass, error)
+	// Get returns the DeviceClass with the given className.
+	Get(className string) (*resourceapi.DeviceClass, error)
+}
+
+// ResourceClaimTracker is used by the DRA plugin to obtain ResourceClaims, and track changes to ResourceClaims in-memory.
+//
+// If the claims are meant to be allocated in the API during the binding phase (when used by scheduler), the tracker helps avoid
+// race conditions between scheduling and binding phases (as well as between the binding phase and the informer cache update).
+//
+// If the binding phase is not used (e.g. when used by Cluster Autoscaler which only runs the scheduling phase, and simulates binding in-memory),
+// the tracker allows the framework user to obtain the claim allocations produced by the DRA plugin, and persist them outside of the API (e.g. in-memory).
+type ResourceClaimTracker interface {
+	// List lists ResourceClaims. The claims can either be obtained from a real informer, or simulated in-memory.
+	//
+	// If a real informer is used, changes made via AssumeClaimAfterAPICall() should be applied to the obtained claims until
+	// they're reflected in the informer cache.
+	//
+	// If claims are meant to be allocated during the binding phase, changes made via SignalClaimPendingAllocation() should
```

Comment:
I'm not sure to what extent those comments are helpful. It feels like instead of describing the expected behavior they describe a reference implementation and rely on the user / implementer to be familiar enough with all the underlying concepts to infer the behavior of said implementation. I'd consider rephrasing the comments here to describe the resulting behavior that a method user should expect and rely on people implementing this interface to figure out how to achieve expected behavior - they can find reference implementation easily enough if they want to.

For example - I'd replace `If a real informer is used, changes made via AssumeClaimAfterAPICall() should be applied to the obtained claims until they're reflected in the informer cache.` with something along the lines of `The result is guaranteed to immediately include any changes made via AssumeClaimAfterAPICall() in order to avoid potential races caused by communication with api-server`.

Reply:
I'd also rephrase other comments in ResourceClaimTracker interface definition in a similar way